core/src/main/scala/kafka/network/RequestChannel.scala

```diff
@@ -205,7 +205,8 @@ class RequestChannel(val numProcessors: Int, val queueSize: Int) extends KafkaMe
 
   /** Send a request to be handled, potentially blocking until there is room in the queue for the request */
   def sendRequest(request: RequestChannel.Request) {
-    requestQueue.put(request)
+    val prioritized = request.requestId == ApiKeys.LEADER_AND_ISR.id || request.requestId == ApiKeys.UPDATE_METADATA_KEY.id || request.requestId == ApiKeys.STOP_REPLICA.id
```

Comment:
Nit: I'd have a set containing the prioritised request ids and then simply use contains instead of having a potentially large number of condition checks.